core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -206,6 +203,14 @@ public void disable() {
         }
 
     }
+    
+    private static class TargetComparator implements Comparator<Target> {
+
+        @Override
+        public int compare(Target left, Target right) {
+            return right.getName().length() - left.getName().length();
+        }
+    }
```

Comment:
The class will be loaded in Static initializer instead of the orderedTargets() call. Static initilizer will be run when the class is first accessed, hence it will happen long before the Recorder starts receiving the data